app/src/main/java/com/salman/roomtask/ui/home/HomeFragment.kt

```diff
@@ -39,13 +49,53 @@ class HomeFragment : Fragment() {
         viewModel.categories.observe(viewLifecycleOwner) { categories ->
             binding.chipGroupCategories.submitList(categories) { category ->
                 // TODO navigate to details fragment with category id
+                findNavController().navigate(R.id.detailsFragment, Bundle().apply {
+                    category.id?.let { putInt("categoryId", it) }
+                    putString("categoryName", category.title)
+                })
             }
         }
 
         viewModel.notes.observe(viewLifecycleOwner) { notes ->
             println(notes)
             adapter.submitList(notes)
+            adapter.setItemClickListener(object : NoteAdapter.OnItemClickListener{
+                override fun onNoteClickListener(note: Note) {
+                    buildSelectCategoryDialog(note)
+                    dialog.show()
+                }
+
+            })
+        }
+    }
+
+    private fun buildSelectCategoryDialog(note: Note) {
```

Comment:
Could you create a new `DialogFragment` instead of this function and navigate to it? just to separate concerns simplify the `HomeFragment`